$NetBSD$

Modifications for "--enable-libtool".

Remove OS specific CXXFLAGS and PANGOLIBDIR selection
(tested to fail on GNU/Linux).

--- configure.ac.orig	2019-04-07 09:24:20.000000000 +0000
+++ configure.ac
@@ -217,7 +217,20 @@ fi
 AC_ARG_ENABLE(gl, [  --enable-gl             turn on OpenGL support [[default=yes]]])
 
 AC_ARG_ENABLE(shared, [  --enable-shared         turn on shared libraries [[default=no]]])
-if test x$enable_shared = xyes; then
+
+# Check whether --enable-libtool or --disable-libtool was given.
+if test "${enable_libtool+set}" = set; then
+  enableval="$enable_libtool"
+
+if test x$enable_libtool != xno; then
+    LIBTOOL="$enable_libtool"
+    enable_shared=no
+else
+    LIBTOOL=""
+fi
+fi
+
+if test "$enable_shared" = "yes"; then
     PICFLAG=1
     SHAREDSUFFIX=""
     FLUID="fluid-shared"
@@ -390,6 +403,21 @@ else
     LINKSHARED="$LINKFLTKCAIRO ../lib/libfltk_images.a ../lib/libfltk_forms.a ../lib/libfltk.a"
 fi
 
+if test "x$LIBTOOL" != "x"; then
+    LINKFLTK="../src/libfltk.la"
+    if test "x$LINKFLTKGL" != "x"; then
+        LINKFLTKGL="../src/libfltk_gl.la"
+    fi
+    if test "x$LINKFLTKIMG" != "x"; then
+        LINKFLTKIMG="../src/libfltk_images.la"
+    fi
+    LINKSHARED="-lfltk_images -lfltk_forms -lfltk"
+    DSONAME="libfltk.la"
+    FLDSONAME="libfltk_forms.la"
+    GLDSONAME="libfltk_gl.la"
+    IMGDSONAME="libfltk_images.la"
+fi
+
 AC_SUBST(DSOCOMMAND)
 AC_SUBST(DSOFLAGS)
 AC_SUBST(DSOLINK)
@@ -1087,25 +1115,27 @@ case $host_os_gui in
 
         pango_found=no
 	if test x$enable_pango = xyes; then
-	    case $host_os in
-    		darwin*)
-	    	  CXXFLAGS="-I/sw/include/pango-1.0 -I/sw/include/glib-2.0 -I/sw/lib/glib-2.0/include $CXXFLAGS"
-	    	  PANGOLIBDIR="-L/sw/lib"
-	    	  ;;
-    		linux*)
-	    	  CXXFLAGS="-I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include $CXXFLAGS"
-	    	  CXXFLAGS="-I/usr/lib/i386-linux-gnu/glib-2.0/include -I/usr/lib/x86_64-linux-gnu/glib-2.0/include $CXXFLAGS"
-    	    	  ;;
-    	    	freebsd*)
-	    	  CXXFLAGS="-I/usr/local/include/pango-1.0 -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include $CXXFLAGS"
-	    	  ;;
-     	    	netbsd*)
-	    	  CXXFLAGS="-I/usr/pkg/include/pango-1.0 -I/usr/pkg/include/glib-2.0 -I/usr/pkg/lib/glib-2.0/include $CXXFLAGS"
-	    	  CXXFLAGS="-I/usr/pkg/include/glib/glib-2.0 $CXXFLAGS"
-	    	  PANGOLIBDIR="-L/usr/pkg/lib"
-	    	  LDFLAGS="-Wl,-rpath,/usr/pkg/lib $LDFLAGS"
-	    	  ;;
-   	    esac
+	dnl # Remove OS specific CXXFLAGS and PANGOLIBDIR selection for pkgsrc
+	dnl # Use ${prefix} instead of hard coded locations
+	dnl     case $host_os in
+    	dnl 	darwin*)
+	dnl     	  CXXFLAGS="-I/sw/include/pango-1.0 -I/sw/include/glib-2.0 -I/sw/lib/glib-2.0/include $CXXFLAGS"
+	dnl     	  PANGOLIBDIR="-L/sw/lib"
+	dnl     	  ;;
+    	dnl 	linux*)
+	dnl     	  CXXFLAGS="-I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include $CXXFLAGS"
+	dnl     	  CXXFLAGS="-I/usr/lib/i386-linux-gnu/glib-2.0/include -I/usr/lib/x86_64-linux-gnu/glib-2.0/include $CXXFLAGS"
+    	dnl     	  ;;
+    	dnl     	freebsd*)
+	dnl     	  CXXFLAGS="-I/usr/local/include/pango-1.0 -I/usr/local/include/glib-2.0 -I/usr/local/lib/glib-2.0/include $CXXFLAGS"
+	dnl     	  ;;
+     	dnl     	netbsd*)
+	    	  CXXFLAGS="-I$prefix/include/pango-1.0 -I$prefix/include/glib-2.0 -I$prefix/lib/glib-2.0/include $CXXFLAGS"
+	    	  CXXFLAGS="-I$prefix/include/glib/glib-2.0 $CXXFLAGS"
+	    	  PANGOLIBDIR="-L$prefix/lib"
+	    	  LDFLAGS="-Wl,-rpath,$prefix/lib $LDFLAGS"
+	dnl     	  ;;
+   	dnl     esac
 	        CPPFLAGS="$PANGOLIBDIR $CXXFLAGS"
 	        AC_CHECK_HEADERS([pango/pango.h pango/pangoxft.h],
 		   AC_CHECK_LIB(pango-1.0, pango_layout_new,
